src/main/java/Alice.java

```diff
@@ -0,0 +1,44 @@
+import storage.Storage;
+import storage.TaskList;
+import ui.Ui;
+
+/**
+ * The entry point to the application.
+ * Encapsulates a TaskList for storing tasks, a Storage for storing and loading
+ * input data, and a Ui for handling user interaction.
+ */
+public class Alice {
+    private final TaskList tasks;
+    private final Storage storage;
+    private final Ui ui;
+
+    /**
+     * Initialises an instance of Alice chatbot.
+     *
+     * @param filePath the path of the file to save the tasks to.
+     */
+    public Alice(String filePath) {
+        storage = new Storage(filePath);
+        // load tasks
+        storage.loadTasks();
+        tasks = storage.getTasks();
+
+        // initialise UI with the loaded tasks
+        ui = new Ui(tasks);
+    }
+
```

Comment:
I like how you've structured the initialization of Storage, TaskList, and Ui in the constructor—it makes the flow of the setup quite clear. I’m wondering, though, would it be beneficial to extract the task-loading logic into a separate method? It could make the constructor more concise and improve readability. Just a thought for making the code a bit cleaner!